src/SDKs/Azure.Base/data-plane/Azure.Base/ClientPipelineBuilderExtensions.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using Azure.Base.Http;
+using Azure.Base.Http.Pipeline;
+
+namespace Azure
+{
+    public static class ClientPipelineBuilderExtensions
+    {
+        public static HttpPipeline Build(this HttpPipelineOptions options, string componentName, string componentVersion)
+        {
+            if (!options.DisableTelemetry)
+            {
+                options = options.Clone();
+                options.AddHeaders(HttpHeader.Common.CreateUserAgent(componentName, componentVersion, options.ApplicationId));
+            }
+
+            var transport = options.TransportPolicy();
+
+            HttpPipelinePolicy currentPolicy = transport;
+
+            foreach (var section in options.PipelinePolicies.Reverse().GroupBy(registration => registration.Section))
+            {
+                foreach (var pipelineBuilderPipelinePolicy in section)
+                {
+                    currentPolicy = pipelineBuilderPipelinePolicy.PolicyFactory(currentPolicy);
+                }
+            }
+
+            return new HttpPipeline(transport, currentPolicy);
+        }
+
+        public static HttpPipelineOptions AddLogging(this HttpPipelineOptions options, int[] excludeCodes = null)
```

Comment:
Why do we need this? The current architectural guidelines say that logging will be on by default.

Reply:
I think this comes down to client overriding customer or customer overriding client problem. You both want to be able for a customer to override client and for a client to override customer (add default retry/logging) I don't think that having both is reasonable it makes things way too complicated because there is no one source of truth.